Detections/SigninLogs/AnomalousIPUsageFollowedByTeamsAction.kql

```diff
@@ -0,0 +1,104 @@
+  - Persistance
+relevantTechniques:
+  - T1199
+  - T1136
+  - T1078
+  - T1098
+query: |
+
+  //The bigger the window the better the data sample size, as we use IP prevalence, more sample data is better.
+  let timeRange = 7d;
+  //The minimum number of countries that the account has been accessed from [default: 2]
+  let minimumCountries = 2;
+  //The delta (%) between the largest in-use IP and the smallest [default: 90]
+  let deltaThreshold = 90;
+  //The maximum (%) threshold that the country appears in login data [default: 10]
+  let countryPrevalenceThreshold = 10;
+  //The time to project forward after the last login activity [default: 60min]
+  let projectedEndTime = 60min; 
+  // Collect successful signins to teams
+  SigninLogs
+  | where TimeGenerated >= ago(timeRange)
+  | where AppDisplayName contains "Teams"
+  | where ConditionalAccessStatus =~ "success"
+  //Uncomment to only look for accounts which used single factor authentication
+  //| where AuthenticationRequirement =~ "singlefactorauthentication"
+  //Uncomment to limit domain, use this on very large tenants to reduce match volume
+  //| where UserPrincipalName endswith "@onmicrosoft.com"
+  | where IPAddress != ""
+  //Get the country information for the successful login
+  | extend country = LocationDetails['countryOrRegion']
+  | summarize count(), any(tostring(country)), StartTime=min(TimeGenerated), EndTime=max(TimeGenerated) by IPAddress, UserPrincipalName
+  //Make a JSON object that contains the summarised IP login evens, including the country the logins were from, start and end time
+  | extend p = pack(tostring(count_), pack("IP",tostring(IPAddress), "Country", any_country, "Start", StartTime, "End", EndTime))
+  //Summarise down the events based on the users email address
+  | summarize make_bag(p), max(count_), min(count_), make_set(any_country), dcount(any_country) by UserPrincipalName
+  //Calculate the delta between the most used IP, and the least used IP. This will find accounts where an IP was used for a small proportion of logins
+  | extend delta = toreal(max_count_ - min_count_) / max_count_ * 100
+  //Where the delta between the largest and smallest IP is >= 90% (configurable) and more than one country is seen (optional)
+  | where delta >= deltaThreshold and dcount_any_country >= minimumCountries
+  //Extract the country code for the IP with the smallest number of logins. To further reduce hits we will use this to check for 
+  //how common that country is as a login location for our tenant.
+  | extend country = tostring(bag_p.[tostring(min_count_)].["Country"])
+  | join (
+      //Get all successful login events to teams from each country
+      SigninLogs
+      | where TimeGenerated > ago(timeRange)
+      | where AppDisplayName contains "Teams"
+      | where ConditionalAccessStatus =~ "success"
+      | extend country = LocationDetails['countryOrRegion']
+      | where country != ""
+      | summarize count() by tostring(country)
```

Comment:
drop tostring() as it should be handled above

Reply:
Fixed